docs/src/main/asciidoc/spring-cloud-sleuth.adoc

```diff
@@ -191,7 +191,120 @@ adding a Channel Binder implementation
 (e.g. `spring-cloud-starter-stream-rabbit` for RabbitMQ or
 `spring-cloud-starter-stream-kafka` for Kafka). This will
 automatically turn your app into a producer of messages with payload
-type `Spans`. 
+type `Spans`.
+
+== Customizations
+
+Thanks to the `SpanInjector` and `SpanExtractor` you can customize the way spans
+are created and propagated.
+
+Essentially with Spring Cloud Span there are two ways span tracing data can be passed to
+ a process.
+
+ * via Spring Integration
+ * via HTTP
+
+=== Spring Integration
+
+For Spring Integration these are the beans responsible for creation of a Span from a `Message`
+ and filling in the `MessageBuilder` with tracing information.
+
+[source,java]
+----
+	@Bean
+	@Qualifier("messagingSpanExtractor")
+	@ConditionalOnProperty(value = "spring.sleuth.integration.extractor.enabled", matchIfMissing = true)
+	public SpanExtractor<Message> messagingSpanExtractor() {
+		...
+	}
+
+	@Bean
+	@Qualifier("messagingSpanInjector")
+	@ConditionalOnProperty(value = "spring.sleuth.integration.injector.enabled", matchIfMissing = true)
+	public SpanInjector<MessageBuilder> messagingSpanInjector() {
+		...
+	}
+----
+
+As you can see you can disable them with the properties:
+
+* `Message` span extractor - `spring.sleuth.integration.extractor.enabled`
+* `MessageBuilder` span injector - `spring.sleuth.integration.injector.enabled`
+
+There is also a WebSockets version:
+
+* `Message` span extractor - `spring.sleuth.integration.websocket.extractor.enabled`
+* `MessageBuilder` span injector - `spring.sleuth.integration.websocket.injector.enabled`
+
+You can override them either by disabling with a property and providing your own implementation
+or by adding a `@Primary` annotation to your bean definition. Remember to provide the following qualifiers:
+
+* `Message` span extractor - `messagingSpanExtractor`
```

Comment:
> this seems like documenting a TODO :) are we sure these qualifiers aren't needed? or do we just think we should document how things are today in case they don't improve?

yes - if someone wants to override the behaviour he needs to pass the qualifier. We can add annotations instead. @MessagingSpanExtractor or sth like that (actually that one is already reserved for the impl).

> are these needed bc messaging, stomp, etc might all be needed in the same spring context? maybe make or reuse a qualifier annotation? just thoughts as strings like this seem fragile

There was an issue with configs - I fixed that